components/RandomDog.js

```diff
@@ -0,0 +1,57 @@
+import React, { useEffect, useState } from 'react'
+import styled from 'styled-components/native'
+import { Image } from 'react-native'
+import { ShareDog } from './ShareDog'
+
+const DogView = styled.View`
+    flex: 1;
+    width: 100%;
+    align-items: center;
+    ` 
+
+const ButtonView = styled.View`
+    flex: 1.5;
+    `
+
+const RandomDogButton = styled.TouchableOpacity`
+    background-color: pink;
+    border-radius: 50px;
+    padding: 10px;
+    margin-top: 30px
+    `
+
+const ButtonText = styled.Text`
+    font-size: 20px;
+    ` 
+
+export const RandomDog = () => {
+    const [dogs, setDogs] = useState({})      
+    
+    const fetchDogs = () => {
+        fetch('https://dog.ceo/api/breeds/image/random')
+        .then((response) => response.json())
+        .then((json) => { 
+            console.log(json)
+            setDogs(json)
+        })
+     }   
+     /* Here we call the function so that a dog is shown when the page is mounted */
+        useEffect(() => {
+            fetchDogs()
+        }, [] )
+    
+         console.log(dogs)   
+        return (
+            <>
+                <DogView>
+                    <Image source={{uri: dogs.message}} style= {{width: 270, height: 270}}/>
```

Comment:
You should rather pass those styles (`width` and `height`) as in styled component

Reply:
I don't know how to do that since this is an image component imported from react-native. I can't use the name Image as name in a styled component. Do you know how I can pass the styles as a styled component?

Reply:
Yep, you can do it like this for example :

```
const DogImage = styled.Image`
    width: 270px;
    height: 270px;
`
```

Reply:
Ah of course, thanks! Didn't think I could change that Image name since it was imported, but the problem is fixed now :)